internal/resources/app_endpoint_activation_status.go

```diff
@@ -0,0 +1,368 @@
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &AppEndpointActivationStatus{}
+	_ resource.ResourceWithConfigure   = &AppEndpointActivationStatus{}
+	_ resource.ResourceWithImportState = &AppEndpointActivationStatus{}
+)
+
+const errorMessageWhileAppEndpointActivation = "There is an error during switching online/offline the app endpoint. Please check in Capella to see if any hanging resources have been created, unexpected error: "
+
+// AppEndpointActivationStatus manages activation status (online/offline) of an App Endpoint.
+type AppEndpointActivationStatus struct {
+	*providerschema.Data
+}
+
+// NewAppEndpointActivationStatus is a helper function to simplify the provider implementation.
+func NewAppEndpointActivationStatus() resource.Resource {
+	return &AppEndpointActivationStatus{}
+}
+
+// ImportState imports a remote App Endpoint activation status that is not created by Terraform.
+func (r *AppEndpointActivationStatus) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Save import name to app_endpoint_name attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("app_endpoint_name"), req, resp)
+}
+
+// Metadata returns the resource type name.
+func (r *AppEndpointActivationStatus) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_app_endpoint_activation_status"
+}
+
+// Schema defines the schema for App Endpoint Activation Status.
+func (r *AppEndpointActivationStatus) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = AppEndpointActivationStatusSchema()
+}
+
+// Configure sets provider-defined data, clients, etc.
+func (r *AppEndpointActivationStatus) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	r.Data = data
+}
+
+// Create switches the app endpoint to online/offline state.
+func (r *AppEndpointActivationStatus) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.AppEndpointActivationStatus
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := r.validate(plan); err != nil {
+		resp.Diagnostics.AddError(
+			"Error parsing create app endpoint activation status request",
+			"Could not switch app endpoint online/offline, unexpected error: "+err.Error(),
+		)
+		return
+	}
```

Comment:
validation should be done at schema level.  see this for an example:
https://github.com/couchbasecloud/terraform-provider-couchbase-capella/blob/main/internal/resources/free_tier_cluster_on_off_schema.go#L9